components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/util/OAuth2UtilTest.java

```diff
@@ -1799,4 +1966,300 @@ public void testValidateRequestTenantDomain(boolean isTenantQualifiedURLsEnabled
+
+        AppInfoCache appInfoCache = mock(AppInfoCache.class);
+        when(appInfoCache.getValueFromCache(clientId)).thenThrow(IdentityOAuth2Exception.class);
+
+        mockStatic(AppInfoCache.class);
+        when(AppInfoCache.getInstance()).thenReturn(appInfoCache);
+
+        try {
+            OAuth2Util.getOAuthTokenIssuerForOAuthApp(clientId);
+        } catch (IdentityOAuth2Exception ex) {
+            assertEquals(ex.getMessage(), "Error while retrieving app information for clientId: " + clientId);
+        }
+    }
+
+    @Test
+    public void testGetClientSecret() throws Exception {
+
+        OAuthAppDO appDO = new OAuthAppDO();
+        appDO.setOauthConsumerKey(clientId);
+        appDO.setOauthConsumerSecret(clientSecret);
+
+        AppInfoCache appInfoCache = mock(AppInfoCache.class);
+        when(appInfoCache.getValueFromCache(clientId)).thenReturn(appDO);
+
+        mockStatic(AppInfoCache.class);
+        when(AppInfoCache.getInstance()).thenReturn(appInfoCache);
+
+        assertEquals(OAuth2Util.getClientSecret(clientId), appDO.getOauthConsumerSecret());
+    }
+
+    @Test
+    public void testGetClientSecretWithException() throws Exception {
+
+        AppInfoCache appInfoCache = mock(AppInfoCache.class);
+        when(appInfoCache.getValueFromCache(clientId)).thenReturn(null);
+
+        mockStatic(AppInfoCache.class);
+        when(AppInfoCache.getInstance()).thenReturn(appInfoCache);
+
+        OAuthAppDAO oAuthAppDAO = mock(OAuthAppDAO.class);
+        whenNew(OAuthAppDAO.class).withNoArguments().thenReturn(oAuthAppDAO);
+        when(oAuthAppDAO.getAppInformation(anyString())).thenReturn(null);
+
+        try {
+            OAuth2Util.getClientSecret(clientId);
+        } catch (InvalidOAuthClientException ex) {
+            assertEquals(ex.getMessage(), "Unable to retrieve app information for consumer key: "
+                    + clientId);
+        }
+    }
+
+    @DataProvider(name = "authenticatedIDPProvider")
+    public static Object[][] authenticatedIDPProvider() {
+
+        String authenticatedIDP = "dummyAuthenticatedIDP";
+        return new Object[][]{
+                {authenticatedIDP, OAuthConstants.UserType.FEDERATED_USER_DOMAIN_PREFIX +
+                        OAuthConstants.UserType.FEDERATED_USER_DOMAIN_SEPARATOR + authenticatedIDP},
+                {null, OAuthConstants.UserType.FEDERATED_USER_DOMAIN_PREFIX}};
+    }
+
+    @Test(dataProvider = "authenticatedIDPProvider")
+    public void testGetFederatedUserDomain(String authenticatedIDP, String expected) {
+
+        assertEquals(OAuth2Util.getFederatedUserDomain(authenticatedIDP), expected);
+    }
+
+    @Test
+    public void testGetTokenBindingReference() {
+
+        assertNotNull(OAuth2Util.getTokenBindingReference("dummyReference"));
+    }
+
+    @Test
+    public void testGetTokenBindingReferenceWithNullValue() {
+
+        assertNull(OAuth2Util.getTokenBindingReference(null));
```

Comment:
Better to write a test case for `OAuth2Util.getTokenBindingReference("")` as well